sam-ruleset-lookup-tool/docs/ruleset_lookup_tool_usage.md

```diff
@@ -0,0 +1,299 @@
+                description: "GDPR data handling rules"
+                decision_logic: |
+                  IF data_type == 'personal':
+                    REQUIRE explicit_consent AND data_protection_officer_approval
+                  IF data_retention > 24_months:
+                    REQUIRE legal_basis_documentation
+          americas:
+            decision_trees:
+              - name: "SOX Compliance"
+                description: "Sarbanes-Oxley compliance rules"
+                decision_logic: |
+                  IF financial_data == true:
+                    REQUIRE dual_approval AND audit_trail
+```
+
+## Tool Behavior
+
+### Dynamic Tool Naming
+The tool generates its function name as: `get_rulesets_{sanitized_name}`
+
+For example:
+- `name: "hr_policies"` → `get_rulesets_hr_policies`
+- `name: "Compliance Rules"` → `get_rulesets_compliance_rules`
+
+### Group Selection Logic
+1. **Exact Match**: Looks for exact group key (case-insensitive)
+2. **Fallback**: Uses `default_group_key` if specified group not found
+3. **Error**: Returns error if no group found and no default configured
+
+### Response Format
+```json
+{
+  "decision_logic": "Ruleset Name: Leave Policy\nDescription: Annual leave calculation rules\nLogic:\nIF employment_years >= 5:\n  RETURN 25 + (employment_years - 5) * 0.5 days (max 30)\nELSE:\n  RETURN 20 + employment_years days\n\n---\n\n..."
+}
+```
+
+## Best Practices
+
+### 1. Clear Logic Structure
+```
+✅ Good:
+IF condition1 AND condition2:
+  RETURN specific_result
+ELSE IF condition3:
+  RETURN alternative_result
+ELSE:
+  RETURN default_result
+
+❌ Avoid:
+Complex nested logic without clear structure
+```
+
+### 2. Descriptive Names
+```yaml
+✅ Good:
+name: "CLA (Collective Labour Agreement)"
+description: "Determines applicable CLA based on employee profile"
+
+❌ Avoid:
+name: "Rule1"
+description: "Some rule"
+```
+
+### 3. Comprehensive Groups
+```yaml
+✅ Good:
+decision_tree_groups:
+  netherlands: { ... }
+  germany: { ... }
+  _default: { ... }  # Always include default
+
+❌ Avoid:
+decision_tree_groups:
+  netherlands: { ... }
+  # No default fallback
+```
+
+### 4. Validation-Friendly Logic
+```
+✅ Good:
+IF employeeGroup == 'internal employee' AND jobGrade IN ['CT1', 'CT2']:
+  RETURN 'NLD_LOC_C&T'
+
+❌ Avoid:
+IF employee is coffee & tea worker:
+  RETURN some_label
+```
+
+## Integration with Agent Instructions
+
+When using this tool in an agent, structure your instructions to:
+
+1. **Identify** when to use the tool
+2. **Extract** the grouping parameter from context
+3. **Evaluate** the returned logic
+4. **Apply** the results appropriately
+
+Example agent instruction:
+```yaml
+instruction: |
+  When answering policy questions:
+  1. Call get_rulesets_hr_policies with the user's country
+  2. Read the returned decision_logic for the relevant topic
+  3. Apply the IF/ELSE logic using the user's profile data
+  4. Use the result to inform your response
+```
+
+## Troubleshooting
```

Comment:
I am not sure there is great value in any of this. For sure, some of the items should be removed because this looks a lot like AI just wanted 4 items in each group